addons/vs_custom_nodes/2DIterator.gd

```diff
@@ -0,0 +1,62 @@
+tool
+extends VisualScriptCustomNode
+
+func _get_input_value_port_count():
+	return 1
+
+func _get_input_value_port_name(idx):
+	return "rect"
+
+func _get_input_value_port_type(idx):
+	return TYPE_RECT2
+
+func _get_output_sequence_port_count():
+	return 2
+
+func _get_output_sequence_port_text(idx):
+	if idx == 1:
+		return "exit"
+	else:
+		return "each"
+
+func _get_output_value_port_count():
+	return 1
+
+func _get_output_port_text(idx):
+	return "position"
+
+func _get_output_value_port_type(idx):
+	return TYPE_VECTOR2
+
+func _has_input_sequence_port():
+	return true
+
+func _get_text():
+	return "for every (position) in (rect)"
+
+func _get_caption():
+	return "2D Iterator"
```

Comment:
I believe 2D iterator is a really ambiguous name. Something like Rect Iterator or would be more obvious.

Reply:
Personally I find "2D iterator" describes exactly what it does: iterate over a 2d area. Rect2 Iterator would only confuse people who don't know what a Rect2 is.

Reply:
Probably anyways we can change it later. Merging it
:)